core/modules/css/DbInterfaceMySql.h

```diff
@@ -0,0 +1,68 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2019 LSST/AURA.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_QSERV_CSS_DBINTERFACE_H
+#define LSST_QSERV_CSS_DBINTERFACE_H
+
+// System headers
+#include <mutex>
+#include <set>
+
+// Third-party headers
+
+// Qserv headers
+#include "mysql/MySqlConfig.h"
+#include "sql/SqlConnection.h"
+
+namespace lsst {
+namespace qserv {
+namespace css {
+
+
+/// This class is used to connect to tabular data in a mysql database.
+class DbInterfaceMySql {
```

Comment:
Ditto. I would also suggest the in-memory implementation (from `std::list`, etc.) for the base interface to facilitate unit testing.

Reply:
This is not really meant to be the empty chunk interface. Everything else in CSS is key-value, this is meant to be an interface for things better represented by database tables, hence DbInterfaceMySql. I've been holding off making an interface class as it complicates the code I don't have immediate need of it.

Reply:
Hmm, without being able to predict what other data do we need in CSS in tabular format it's hard to tell how useful this tabular data interface is going to be. It may well be that empty chunk list is the only thing that we need.